### ISSUE TITLE
Switch XunitTestAssembly serialization from AssemblyName to AssemblyPath

### ISSUE THREAD

**Issue by estrizhok** [xunit/xunit#2966]
xUnit v3 0.1.1-pre.489

STR:

1. Take any normal assembly & rename it physically on the disk.
2. Run the following test

```
[Fact]
public void SerializationRoundTrip()
{
	var assembly = Assembly.LoadFrom(<path_to_renamed_assembly>);
	var testAssembly = new TestAssembly(new ReflectionAssemblyInfo(assembly));
	var info = new XunitSerializationInfo();

	testAssembly.Serialize(info);

#pragma warning disable CS0618 // Type or member is obsolete
	var copy = new TestAssembly();
#pragma warning restore CS0618 // Type or member is obsolete

	copy.Deserialize(info);
}
```

Expected result:
1. Test passed

Actual result:
1. Test fails with `System.IO.FileNotFoundException : Could not load file or assembly` because it tries to load the assembly by its name (although it has the full path to the assembly)

**Comment by bradwilson** [comment]
Assembly serialization should be based on file name and not assembly name, now that file names are always required, to work around this issue.

**Comment by bradwilson** [comment]
This is the updated test for 0.2.0-pre.8 (from my repro project):

```csharp
[Fact]
public void SerializationRoundTrip()
{
	var assembly = Assembly.LoadFrom(<path_to_renamed_assembly>);
	var testAssembly = new XunitTestAssembly(assembly);
	var info = new XunitSerializationInfo();

	testAssembly.Serialize(info);

#pragma warning disable CS0618 // Type or member is obsolete
	var copy = new XunitTestAssembly();
#pragma warning restore CS0618 // Type or member is obsolete

	copy.Deserialize(info);
}
```

**Comment by bradwilson** [comment]
Fixed in v3 `0.2.0-pre.17` https://xunit.net/docs/using-ci-builds
